components/style/sharing/mod.rs

```diff
@@ -698,13 +698,8 @@ impl<E: TElement> StyleSharingCache<E> {
             return None;
         }
 
-        if target.local_name() != candidate.element.local_name() {
-            trace!("Miss: Local Name");
-            return None;
-        }
-
-        if target.namespace() != candidate.element.namespace() {
-            trace!("Miss: Namespace");
+        if !target.is_same_type(&candidate.element) {
```

Comment:
This doesn't need to change either, and I'd prefer to keep the two checks to remain separate to see why it fails.